File: data/data.code.lua.server/MsgSceneVector3.lua
module("MsgSceneVector3", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = MsgSceneVector3})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.x = 0
	self.y = 0
	self.z = 0
end


function encode(self)
	local pack = packet.new()
	pack:writeI16(self.x)
	pack:writeI16(self.y)
	pack:writeI16(self.z)
	return pack:readBytes()
end

function decode(self, pack)
	self.x = pack:readI16()
	self.y = pack:readI16()
	self.z = pack:readI16()
	return self
end

function getBytes(self)
	return self:encode()
end

return MsgSceneVector3

File: examples/golang/server/proto/msg_scene_vector_3.go
package proto

import (
	"packet"
)

type MsgSceneVector3 struct {
	X                        int16
	Y                        int16
	Z                        int16
}

func (this *MsgSceneVector3) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteInt16(this.X)
	pack.WriteInt16(this.Y)
	pack.WriteInt16(this.Z)

	return pack.ReadBytes()
}

func MsgSceneVector3Decode(pack *packet.Packet) *MsgSceneVector3 {
	msgSceneVector3 := &MsgSceneVector3{}

	msgSceneVector3.X = pack.ReadInt16()
	msgSceneVector3.Y = pack.ReadInt16()
	msgSceneVector3.Z = pack.ReadInt16()
	return msgSceneVector3
}
